builtin/env.c

```diff
@@ -1362,20 +1362,22 @@ void *$eventloop(void *arg) {
                 exit(-1);
             }
             EVENT_del_read(fd);
-        }
-        switch (fd_data[fd].kind) {
+            $init_FileDescriptorData(fd);
```

Comment:
@nordlander I'm thinking about this in particular. close(fd) seems fairly safe / right thing to do, but what about our fd_data, how do we say that a particular fd slot is unused?

Reply:
Setting `fd_data[fd].kind = nohandler` is the proper way to indicate that the slot is unused. But we should also null out the closure pointer fields for the benefit of our future GC (that's `chandler`, `rhandler`, `errhandler` and `conn`). What we do with the remaining fields is irrelevant, as far as I can see.